src/OpenTK.GLControl/WinGLControl.cs

```diff
@@ -51,53 +49,8 @@ public IGraphicsContext CreateContext(int major, int minor, GraphicsContextFlags
             return new GraphicsContext(mode, WindowInfo, major, minor, flags);
         }
 
-        public bool IsIdle => !PeekMessage(ref msg, IntPtr.Zero, 0, 0, 0);
+        public bool IsIdle => !User32.Message.PeekMessage(out message, IntPtr.Zero, 0, 0, 0);
```

Comment:
Are you sure that message are allocated in managed code before call? Last time got issue by passing huge struct with `out` mod(it was nested and with arrays) and it genereated runtime exceptions, always with random time after call while trying to Marshal struct, changes to ref with preallocation before call fixed issue.

Reply:
Since `message` is a field, it will _always_ be initialized before we access it.
This means that 
`private Msg message;` is (at least for normal usage) equivalent to
`private Msg message = default(Msg);`

So yes, I'm pretty sure that the message is allocated before the unmanaged function call.

Reply:
Ok : )

Reply:
What if someone make call:
Msg msg;
PeekMessage(out msg);

Reply:
`msg´ will be initialized to its default value. Even then, with an out, you don't strictly need to initialize anything since out guarantees initialization. Might want to brush up on that.